src/js/other-websites/inpage_menu.js

```diff
@@ -51,226 +52,610 @@ async function isUserSignedIn() {
-  remainingAliasesSpan.textContent = browser.i18n.getMessage(
-    "popupRemainingAliases_2_mask",
-    [numAliasesRemaining, maxNumAliases]
+  // Add #/# labels to search filter
+  const searchFilterTotal = document.querySelector(".js-filter-masks-total");
+  const searchFilterVisible = document.querySelector(
+    ".js-filter-masks-visible"
   );
 
-  // Free user (who once was premium): Set text informing them how they have exceeded the maximum amount of aliases and cannot create any more
-  if (numAliasesRemaining < 0) {
-    remainingAliasesSpan.textContent = browser.i18n.getMessage(
-      "pageFillRelayAddressLimit_mask"
-    );
-  }
+  searchFilterVisible.textContent = maskResults.length;
 
-  // Premium user: Set text informing them how many aliases they have created so far
-  if (premium) {
-    remainingAliasesSpan.textContent = browser.i18n.getMessage(
-      "popupUnlimitedAliases_mask",
-      [relayAddresses.length]
-    );
-  }
+  searchFilterVisible.textContent = maskResults.filter(
+    (maskResult) => !maskResult.classList.contains("is-hidden")
+  ).length;
+  searchFilterTotal.textContent = maskResults.length;
+}
 
-  const maxNumAliasesReached = numAliasesRemaining <= 0;
+const buildContent = {
+  loggedIn: {
+    free: async (relaySiteOrigin) => {
+      const fxRelayMenuBody = document.getElementById("fxRelayMenuBody");
 
-  // Create "Manage All Aliases" link
-  const relayMenuDashboardLink = document.querySelector(
-    ".fx-relay-menu-dashboard-link"
-  );
-  relayMenuDashboardLink.textContent = browser.i18n.getMessage("ManageAllAliases_mask");
-  relayMenuDashboardLink.href = `${relaySiteOrigin}?utm_source=fx-relay-addon&utm_medium=input-menu&utm_content=manage-all-addresses`;
-  relayMenuDashboardLink.addEventListener("click", () => {
-    sendInPageEvent("click", "input-menu-manage-all-aliases-btn");
-    iframeCloseRelayInPageMenu();
-  });
+      const signedInContentFree = document.querySelector(
+        ".fx-content-signed-in-free"
+      );
 
-  // Create "Get unlimited aliases" link
-  getUnlimitedAliasesBtn.href = `${relaySiteOrigin}/premium?utm_source=fx-relay-addon&utm_medium=input-menu&utm_content=get-premium-link`;
-  getUnlimitedAliasesBtn.addEventListener("click", () => {
-    sendInPageEvent("click", "input-menu-get-premium-btn");
-    iframeCloseRelayInPageMenu();
-  });
+      const signedInContentPremium = document.querySelector(
+        ".fx-content-signed-in-premium"
+      );
+
+      // Create "You have .../.. remaining relay address" message
+      const remainingAliasesSpan = document.querySelector(
+        ".fx-relay-menu-remaining-aliases"
+      );
+
+      const { maxNumAliases } = await browser.storage.local.get(
+        "maxNumAliases"
+      );
+
+      const maskLists = document.querySelectorAll(".fx-relay-menu-masks-list");
+      const masks = await getMasks();
+
+      if (masks.length !== 0) {
+        //
+      }
```

Comment:
TODO: Remove code, or possibly add some logic when the user has zero masks created.